1-js-and-async-programming/1-javascript-fundamentals/3-factorial-chain.js

```diff
@@ -31,6 +31,19 @@ Requirements:
 
 */
 
-const factorialChain = (number, lastDigits) => {};
-
-module.exports = factorialChain;
+const factorialChain = (number, lastDigits) => {
+    if(number < 0 || lastDigits < 0) throw Error('Only possitive numbers are accept');
+    let factorialSum = 0;
+    let factorial = 1;
+    for(let i = 1; i <= number ; i++){
+        factorial = factorial * i;
+        factorialSum += acc;
+    };
+    let stringSum = String(factorialSum);
+    while(stringSum.length < lastDigits){
+        stringSum = '0' + stringSum;
+    }
+    return stringSum.slice(-lastDigits);
```

Comment:
Leverage the built-in functions of a String type
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice

```suggestion
    const stringedFactorial = String(factorialSum)
    const mappedFactorial = 
      size >= stringedFactorial.length ? 
        stringedFactorial.padStart(size, '0') :
        stringedFactorial.slice(-size)

   return mappedFactorial
```

Reply:
That's actually a better option than manually iterating